### PR TITLE
[libmediainfo] Add new port

### PR DESCRIPTION
**Pull request by g-maxime**
[MediaInfoLib](https://mediaarea.net/en/MediaInfo) is a library used for retrieving technical information and other metadata about audio or video files.

Depends on:  #7004

### CODE REVIEW

File: ports/libmediainfo/CONTROL [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Source: libmediainfo
+Version: 19.04
```
Comment (LilyWangL) [inline ports/libmediainfo/CONTROL]: Can you please add the Homepage for this port in the CONTROL file?
```
Homepage: https://github.com/MediaArea/MediaInfoLib
```

File: ports/libmediainfo/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,28 @@
+include(vcpkg_common_functions)
```
Comment (LilyWangL) [inline ports/libmediainfo/portfile.cmake]: `include(vcpkg_common_functions)` is deprecated. Please remove it.

File: ports/libmediainfo/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO MediaArea/MediaInfoLib
+    REF v19.07
+    SHA512 733f660611fc0aec15b4825022d46a29611f7faa113f8910057f31b3524839aa0aa2ffbefb734385b87a500c5cb9cb2cfae6545a0d382fff9a843a472ce13f25
+    HEAD_REF master
+
```
Comment (LilyWangL) [inline ports/libmediainfo/portfile.cmake]: Please remove the extra space lines in portfile.cmake.

File: ports/libmediainfo/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+vcpkg_install_cmake()
+vcpkg_fixup_cmake_targets(CONFIG_PATH share/mediainfolib TARGET_PATH share/mediainfolib)
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/lib/pkgconfig)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig)
+
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/libmediainfo RENAME copyright)
+
+vcpkg_test_cmake(PACKAGE_NAME MediaInfoLib MODULE)
```
Comment (LilyWangL) [inline ports/libmediainfo/portfile.cmake]: `vcpkg_test_cmake` is deprecated. Please remove it.

**Review by LilyWangL — CHANGES_REQUESTED** [review]

**Comment by g-maxime** [comment]
@LilyWangL PR updated, thank for your review.

But I'm unable to reproduce the linker error locally.

**Comment by LilyWangL** [comment]
@g-maxime Currently, this PR fails on Windows due to:
```
CMakeFiles\mediainfo.dir\C_\vsts\_work\3\s\buildtrees\libmediainfo\src\v19.09-d26eded3aa\Source\ThirdParty\tinyxml2\tinyxml2.cpp.obj -LIBPATH:C:\vsts\_work\3\s\installed\x64-windows\debug\lib -LIBPATH:C:\vsts\_work\3\s\installed\x64-windows\share\zenlib\..\..\debug\lib C:\vsts\_work\3\s\installed\x64-windows\debug\lib\libcurl-d.lib wldap32.lib winmm.lib ws2_32.lib advapi32.lib crypt32.lib advapi32.lib OpenSSL::SSL.lib OpenSSL::Crypto.lib C:\vsts\_work\3\s\installed\x64-windows\debug\lib\zlibd.lib crypt32.lib C:\vsts\_work\3\s\installed\x64-windows\share\zenlib\..\..\debug\lib\zend.lib C:\vsts\_work\3\s\installed\x64-windows\debug\lib\zlibd.lib advapi32.lib OpenSSL::SSL.lib OpenSSL::Crypto.lib C:\vsts\_work\3\s\installed\x64-windows\debug\lib\zlibd.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib 
LINK : fatal error LNK1104: cannot open file 'OpenSSL::SSL.lib'
```

**Comment by LilyWangL** [comment]
@g-maxime Could you modify **CMakeLists.txt L90** to `find_package(CURL CONFIG REQUIRED)`. Using curl which build on VCPKG instead of system curl.

**Comment by g-maxime** [comment]
@LilyWangL Done, but still failing.

**Comment by JackBoosY** [comment]
This regression may fix in #9613.

File: ports/libmediainfo/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+)
+
+vcpkg_install_cmake()
+vcpkg_fixup_cmake_targets(CONFIG_PATH share/mediainfolib TARGET_PATH share/mediainfolib)
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/lib/pkgconfig)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig)
+
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/libmediainfo RENAME copyright)
```
Comment (NancyLi1013) [inline ports/libmediainfo/portfile.cmake]: Could you please update 
`file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/libmediainfo RENAME copyright)`
as
`file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)`
?

File: ports/libmediainfo/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    PREFER_NINJA
+    OPTIONS -DBUILD_ZENLIB=0 -DBUILD_ZLIB=0
+)
+
+vcpkg_install_cmake()
+vcpkg_fixup_cmake_targets(CONFIG_PATH share/mediainfolib TARGET_PATH share/mediainfolib)
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/lib/pkgconfig)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig)
```
Comment (NancyLi1013) [inline ports/libmediainfo/portfile.cmake]: The files in `lib/pkgconfig` and `debug/lib/pkgconfig` should not be removed. since needed on Linux platform.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by JackBoosY** [comment]
Seems have regressions in `x86-windows` / `x64-windows` / `arm64-windows`, please get logs [here](https://dev.azure.com/vcpkg/public/_build/results?buildId=30795&view=artifacts&type=publishedArtifacts).

**Comment by g-maxime** [comment]
@JackBoosY I known but I'm unable to reproduce this bug locally and openSSL isn't a direct dependency of libmediainfo.

**Comment by JackBoosY** [comment]
```
  Target "mediainfo" links to target "OpenSSL::SSL" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
Call Stack (most recent call first):
  CMakeLists.txt:359 (add_library)
This warning is for project developers.  Use -Wno-dev to suppress it.
```

Seems missing dependency `openssl`? I added it. Let's see the result.

**Comment by JackBoosY** [comment]
error:
```
LINK : fatal error LNK1104: cannot open file 'OpenSSL::SSL.lib'
```

**Comment by LilyWangL** [comment]
Pinging @g-maxime for response. Is work still being done for this PR?

**Comment by LilyWangL** [comment]
The CI Linux error will be fixed in PR #11238.

**Comment by LilyWangL** [comment]
Hi @g-maxime, this port build failed on `arm64-windows` with the following error:
```
CMakeFiles\mediainfo.dir\C_\agent\_work\1\s\buildtrees\libmediainfo\src\v20.03-187adc44c5\Source\ThirdParty\tinyxml2\tinyxml2.cpp.obj C:\agent\_work\1\s\installed\arm64-windows\debug\lib\libcurl-d.lib wldap32.lib winmm.lib ws2_32.lib advapi32.lib crypt32.lib advapi32.lib C:\agent\_work\1\s\installed\arm64-windows\debug\lib\libssl.lib C:\agent\_work\1\s\installed\arm64-windows\debug\lib\libcrypto.lib C:\agent\_work\1\s\installed\arm64-windows\debug\lib\zlibd.lib C:\agent\_work\1\s\installed\arm64-windows\share\zenlib\..\..\debug\lib\zend.lib C:\agent\_work\1\s\installed\arm64-windows\debug\lib\zlibd.lib crypt32.lib C:\agent\_work\1\s\installed\arm64-windows\debug\lib\zlibd.lib kernel32.lib user32.lib 
   Creating library mediainfod.lib and object mediainfod.exp
aes_modes.c.obj : error LNK2019: unresolved external symbol aes_encrypt referenced in function aes_ecb_encrypt
aes_modes.c.obj : error LNK2019: unresolved external symbol aes_decrypt referenced in function aes_ecb_decrypt
aeskey.c.obj : error LNK2019: unresolved external symbol aes_encrypt_key128 referenced in function aes_encrypt_key
aeskey.c.obj : error LNK2019: unresolved external symbol aes_encrypt_key192 referenced in function aes_encrypt_key
aeskey.c.obj : error LNK2019: unresolved external symbol aes_encrypt_key256 referenced in function aes_encrypt_key
aeskey.c.obj : error LNK2019: unresolved external symbol aes_decrypt_key128 referenced in function aes_decrypt_key
File__Analyze.cpp.obj : error LNK2001: unresolved external symbol aes_decrypt_key128
aeskey.c.obj : error LNK2019: unresolved external symbol aes_decrypt_key192 referenced in function aes_decrypt_key
aeskey.c.obj : error LNK2019: unresolved external symbol aes_decrypt_key256 referenced in function aes_decrypt_key
mediainfod.dll : fatal error LNK1120: 8 unresolved externals
ninja: build stopped: subcommand failed.
```

**Review by NancyLi1013 — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by strega-nil** [comment]
Cool, this LGTM! Thanks @g-maxime :)

**Pull request merged**